### ISSUE TITLE
Handle duplicates in the `companies` and `co2` datasets

### ISSUE THREAD

**Issue by Tilmon** [2DegreesInvesting/tiltIndicator#230]
Dear @maurolepore and @kalashsinghal , 

running the pctr with all real data works for `pctr_at_product_level()` but not for `pctr_at_company_level()`. It throws me the following error: 

Do you need to see the data I used? The pctr_companies file is 500MB large, would probably need to upload it on Google Drive. 

```{r}
dt_test_products <- companies |>  pctr_at_product_level(co2)

dt_test_company <- dt_test_products |> pctr_at_company_level() 
Warning: Values from `risk_category` are not uniquely identified; output will contain list-cols.
• Use `values_fn = list` to suppress this warning.
• Use `values_fn = {summary_fun}` to summarise duplicates.
• Use the following dplyr code to identify duplicates.
  {data} %>%
  dplyr::group_by(company_id, activity_uuid_product_uuid, ei_activity_name, unit.x, unit.y, tilt_sector, isic_4digit_sector,
  co2_footprint, perc_all, perc_unit, perc_tilt_sec, perc_isic_sec, perc_unit_tilt_sec, perc_unit_isic_sec, grouped_by) %>%
  dplyr::summarise(n = dplyr::n(), .groups = "drop") %>%
  dplyr::filter(n > 1L)Error in `fn()`:
! Can't join `x$score` with `y$score` due to incompatible types.
ℹ `x$score` is a <character>.
ℹ `y$score` is a <list>.
Backtrace:
  1. tiltIndicator::pctr_at_company_level(dt_test_products)
  2. tiltIndicator:::xctr_at_company_level(...)
 11. tiltIndicator:::xctr_at_company_level_impl(...)
 12. purrr::reduce(...)
 13. purrr:::reduce_impl(.x, .f, ..., .init = .init, .dir = .dir)
 15. dplyr:::left_join.data.frame(out, elt, ...)

```

Pls find the real data [here ](https://drive.google.com/drive/u/0/folders/15AsbEhET2WvU0z2dFhyKfvZF9jQHPuvP))(only accessible for people with EI license). 
Code that I used to read in data: 

```{r}
pctr_companies <- read_csv(here("input_data/ctr_companies.csv"))
companies <- pctr_companies |> 
  rename(company_id = companies_id) |> 
  select(-clustered)

pctr_ecoinvent_co2 <- read_csv(here("input_data/pctr_products.csv"), 
                               col_types = cols(
                                 isic_4digit = col_number(), 
                                 co2_footprint = col_number()))
co2 <- pctr_ecoinvent_co2 |> 
  rename(isic_4digit_sector = isic_4digit) |> 
  select(-ei_activity_name, -tilt_subsector)

```

**Comment by Tilmon** [comment]
@maurolepore addendum: When reading in the files on my machine, the variable `isic_4digit ` is classified as character while in the sample data from the package, it's classified as number. Changing that didn't solve the problem but wanted to flag just in case relevant for future development.

**Comment by Tilmon** [comment]
@maurolepore please also note: Right now, one `clustered` of one `companies_id ` can be mapped to multiple rows in Ecoinvent. I don't know how this would affect results but wanted to flag it. I will however resolve this issue asap so that you will always have a 1-to-1 match between `companies_id`-`clustered` and `activity_uuid_product_uuid `

**Comment by maurolepore** [comment]
@Tilmon 

RE 

> the variable isic_4digit  is classified as character while in the sample data from the package, it's classified as number.

Which one you think is correct?:

`isic_4digit`:

1. Character: "0710"
2. Number: 710

**Comment by Tilmon** [comment]
@maurolepore option 1. is correct, i.e. character value "o710"

> Which one you think is correct?:

isic_4digit:

Character: "0710"
Number: 710

**Comment by Tilmon** [comment]
Dear @maurolepore pls find attached a folder with code and data that still shows the same problem. 
[tiltIndicator.zip](https://github.com/2DegreesInvesting/tiltIndicator/files/11409550/tiltIndicator.zip)

**Comment by maurolepore** [comment]
Thanks @Tilmon, 

I can reproduce your problem. It seems that we also need to remove duplicates from the `co2` dataset.

``` r
library(dplyr, warn.conflicts = FALSE)
library(tiltIndicator)

companies <- tibble(company_id = "a", activity_uuid_product_uuid = "x")

co2 <- tribble(
  ~activity_uuid_product_uuid, ~unit, ~tilt_sector, ~isic_4digit, ~co2_footprint,
                          "x",  "kg",           NA,       "0710",              1,             
                          "x",  "kg",           NA,       "0710",              1,             
)

# Fails
try(pctr(companies, co2))
#> Warning: Values from `risk_category` are not uniquely identified; output will contain
#> list-cols.
#> • Use `values_fn = list` to suppress this warning.
#> • Use `values_fn = {summary_fun}` to summarise duplicates.
#> • Use the following dplyr code to identify duplicates.
#>   {data} %>%
#>   dplyr::group_by(company_id, activity_uuid_product_uuid, unit, tilt_sector,
#>   isic_4digit, co2_footprint, perc_all, perc_isic_sec, perc_tilt_sec,
#>   perc_unit, perc_unit_isic_sec, perc_unit_tilt_sec, grouped_by) %>%
#>   dplyr::summarise(n = dplyr::n(), .groups = "drop") %>%
#>   dplyr::filter(n > 1L)
#> Error in fn(out, elt, ...) : 
#>   Can't join `x$score` with `y$score` due to incompatible types.
#> ℹ `x$score` is a <character>.
#> ℹ `y$score` is a <list>.

# Works
pctr(companies, distinct(co2))
#> # A tibble: 18 × 4
#>    companies_id grouped_by    risk_category value
#>    <chr>        <chr>         <chr>         <dbl>
#>  1 a            all           high              1
#>  2 a            all           medium            0
#>  3 a            all           low               0
#>  4 a            isic_sec      high              1
#>  5 a            isic_sec      medium            0
#>  6 a            isic_sec      low               0
#>  7 a            tilt_sec      high              1
#>  8 a            tilt_sec      medium            0
#>  9 a            tilt_sec      low               0
#> 10 a            unit          high              1
#> 11 a            unit          medium            0
#> 12 a            unit          low               0
#> 13 a            unit_isic_sec high              1
#> 14 a            unit_isic_sec medium            0
#> 15 a            unit_isic_sec low               0
#> 16 a            unit_tilt_sec high              1
#> 17 a            unit_tilt_sec medium            0
#> 18 a            unit_tilt_sec low               0
```

<sup>Created on 2023-05-05 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by maurolepore** [comment]
``` r
library(tiltIndicator)
library(readr, warn.conflicts = FALSE)
library(dplyr, warn.conflicts = FALSE)
library(here)
#> here() starts at /home/rstudio/Downloads/ti230

packageVersion("tiltIndicator")
#> [1] '0.0.0.9041'

options(readr.show_col_types = FALSE)

companies <- read_csv(here("tr_companies.csv")) |> 
  rename(company_id = companies_id)
inputs <- read_csv(here("pctr_products.csv")) |> 
  # Ugh :-(
  rename(input_co2_footprint = co2_footprint) |> 
  rename(input_tilt_sector = tilt_sector) |> 
  rename(input_unit = unit) |> 
  rename(input_isic_4digit = isic_4digit) |> 
  # Maybe we should just drop NA's with a warning?
  filter(!is.na(input_co2_footprint))
ictr(companies, inputs)
#> # A tibble: 342 × 4
#>    companies_id                              grouped_by risk_category value
#>    <chr>                                     <chr>      <chr>         <dbl>
#>  1 alteisen-reinhard-gmbh_00000005211750-001 all        high              0
#>  2 alteisen-reinhard-gmbh_00000005211750-001 all        medium            0
#>  3 alteisen-reinhard-gmbh_00000005211750-001 all        low               1
#>  4 alteisen-reinhard-gmbh_00000005211750-001 isic_sec   high              0
#>  5 alteisen-reinhard-gmbh_00000005211750-001 isic_sec   medium            1
#>  6 alteisen-reinhard-gmbh_00000005211750-001 isic_sec   low               0
#>  7 alteisen-reinhard-gmbh_00000005211750-001 tilt_sec   high              0
#>  8 alteisen-reinhard-gmbh_00000005211750-001 tilt_sec   medium            0
#>  9 alteisen-reinhard-gmbh_00000005211750-001 tilt_sec   low               1
#> 10 alteisen-reinhard-gmbh_00000005211750-001 unit       high              0
#> # ℹ 332 more rows

products <- read_csv(here("pctr_products.csv"))
pctr(companies, products)
#> # A tibble: 342 × 4
#>    companies_id                              grouped_by risk_category value
#>    <chr>                                     <chr>      <chr>         <dbl>
#>  1 alteisen-reinhard-gmbh_00000005211750-001 all        high              0
#>  2 alteisen-reinhard-gmbh_00000005211750-001 all        medium            0
#>  3 alteisen-reinhard-gmbh_00000005211750-001 all        low               1
#>  4 alteisen-reinhard-gmbh_00000005211750-001 isic_sec   high              0
#>  5 alteisen-reinhard-gmbh_00000005211750-001 isic_sec   medium            1
#>  6 alteisen-reinhard-gmbh_00000005211750-001 isic_sec   low               0
#>  7 alteisen-reinhard-gmbh_00000005211750-001 tilt_sec   high              0
#>  8 alteisen-reinhard-gmbh_00000005211750-001 tilt_sec   medium            0
#>  9 alteisen-reinhard-gmbh_00000005211750-001 tilt_sec   low               1
#> 10 alteisen-reinhard-gmbh_00000005211750-001 unit       high              0
#> # ℹ 332 more rows
```

<sup>Created on 2023-05-05 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by maurolepore** [comment]
Closed by #245
